File: C++/Recursion/TOH.cpp
#include<iostream>
using namespace std;

void TOH(int n, string source, string helper, string destination){
    if(n == 1){
        cout<<"Shifting disk "<<n<<" from "<<source<<" to "<<destination<<"."<<endl;
        return;
    } 
    else{
        TOH(n-1, source, destination, helper);
        cout<<"Shifting disk "<<n<<" from "<<source<<" to "<<destination<<"."<<endl;
        TOH(n-1, helper, source, destination);
    }
}

int main(){
    Solution s1;
    int arr[] = {1,2,2};;
    //TOH(3, "Source", "Helper", "Destination");
    return 0;
}

File: Java/toh.java
public class toh{
    public static void TOH(int n, String Source, String Helper, String Destination){
        if(n == 1){
            System.out.println("Transfer 1 from "+Source+" to "+Destination);
        }
        else{
            TOH(n-1, Source, Destination, Helper);
            System.out.println("Transfer "+n+" from "+Source+" to "+Destination);
            TOH(n-1, Helper, Source, Destination);
        }
    } 
    public static void main(String args[]) {
        int n = 3;
        TOH(n, "S", "H", "D");
    }
}